pkg/line/line.go

```diff
@@ -229,6 +229,11 @@ func (p *Parser) LineToEvents(line string, sampleErrors prometheus.CounterVec, s
 
 	var samples []string
 	lineParts := strings.SplitN(elements[1], "|", 3)
+	if len(lineParts) < 2 {
+		sampleErrors.WithLabelValues("not_enough_parts_after_colon").Inc()
+		level.Debug(logger).Log("msg", "Bad line (not enough parts after first :) from StatsD", "line", line)
```

Comment:
```suggestion
		level.Debug(logger).Log("msg", "Bad line from StatsD, not enough parts after first ':'", "line", line)
```

Reply:
No, at this point we have already split on `:` and _then_ we don't have enough `|` delimited parts left. I'll try to make the log message clearer.

Reply:
The parsing code is a bit of a mess, partially because of the behavior it currently has where it needs to deal with all flavors at once (#580). It's hard to describe what's wrong with the line because we don't really know, we only know that we've chopped the line in half and now the second half doesn't look like we think it should.

Reply:
Ahh, ok. Makes sense.